client/settings/settings.tsx

```diff
@@ -1,228 +1,391 @@
-    case SettingsTab.Sound:
-      contents = (
-        <SoundSettings
-          scrSettings={tempScrSettings}
-          formRef={formRef}
-          onChange={onSettingsChange}
-          onSubmit={onSettingsSubmit}
-        />
-      )
-      break
-    case SettingsTab.Video:
-      contents = (
-        <VideoSettings
-          scrSettings={tempScrSettings}
-          formRef={formRef}
-          onChange={onSettingsChange}
-          onSubmit={onSettingsSubmit}
-        />
-      )
-      break
-    case SettingsTab.Gameplay:
-      contents = (
-        <GameplaySettings
-          localSettings={tempLocalSettings}
-          scrSettings={tempScrSettings}
-          formRef={formRef}
-          onChange={onSettingsChange}
-          onSubmit={onSettingsSubmit}
-        />
-      )
-      break
-    default:
-      contents = assertUnreachable(activeTab)
-  }
+const ContentContainer = styled.div`
+  width: 100%;
+  max-width: 960px;
+  padding: 16px;
+  overflow-y: auto;
+`
 
+const TitleBar = styled.div`
+  position: relative;
+  margin-bottom: 8px;
+
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+  gap: 16px;
+`
+
+const Title = styled.span`
+  ${headline4};
+`
+
+const CloseIcon = styled(MaterialIcon).attrs({ icon: 'close' })`
+  flex-shrink: 0;
+`
+
+function ContentWrapper({
+  title,
+  children,
+  onCloseSettings,
+}: {
+  title: string
+  children: React.ReactNode
+  onCloseSettings: () => void
+}) {
   return (
-    <Dialog
-      dialogRef={dialogRef}
-      title='Settings'
-      titleAction={titleAction}
-      tabs={tabs}
-      alwaysHasTopDivider={true}
-      buttons={buttons}
-      onCancel={onCancel}>
-      <ContentsBody>
-        {lastError ? (
-          <ErrorText>There was an issue saving the settings. Please try again.</ErrorText>
-        ) : null}
+    <ContentContainer>
+      <TitleBar>
+        <Title>{title}</Title>
+        <Tooltip text='Close settings (ESC)' position='bottom' tabIndex={-1}>
+          <IconButton icon={<CloseIcon />} onClick={onCloseSettings} />
+        </Tooltip>
+      </TitleBar>
 
-        {contents}
-      </ContentsBody>
-    </Dialog>
+      {children}
+    </ContentContainer>
   )
 }
+
+function SettingsContent({
+  subPage,
+  onCloseSettings,
+}: {
+  subPage: SettingsSubPage
+  onCloseSettings: () => void
+}) {
+  const { t } = useTranslation()
+  const label = settingsSubPageToLabel(subPage, t)
+
+  // TODO(2Pac): Currently we don't have any non-Electron settings so this might look a bit weird.
+  // Soon enough we should probably have some user settings (e.g. editing user account, and changing
+  // a language).
+
+  switch (subPage) {
+    case AppSettingsSubPage.Sound:
+      return IS_ELECTRON ? (
+        <ContentWrapper title={label} onCloseSettings={onCloseSettings}>
+          <AppSoundSettings />
+        </ContentWrapper>
+      ) : null
+    case AppSettingsSubPage.System:
+      return IS_ELECTRON ? (
```

Comment:
it also sucks to repeat this check over and over again imo, it would be better to make two switch cases (an `assertUnreachable` at the end of the function when there are two different switch cases that cover all options should still work)

Reply:
That is:

```ts
switch (subPage) {
  // stuff for everything
}

if (!IS_ELECTRON) {
  throw new Error(`Unknown settings page: ${subPage}`)
}

switch (subPage) {
  // stuff for electron
}

return assertUnreachable(subPage)
```